src/python/pants/backend/javascript/nodejs_project.py

```diff
@@ -173,7 +173,15 @@ def full_globs(self) -> Iterable[str]:
 
     def matches_glob(self, pkg_json: PackageJson) -> bool:
         path = PurePath(pkg_json.root_dir)
-        return any(path.match(glob) for glob in self.full_globs())
+
+        def safe_match(glob: str) -> bool:
+            if pkg_json.root_dir == "":
+                return pkg_json.root_dir == glob
+            elif not glob:
+                return False
```

Comment:
Small thing, but I had to read this a few times to really understand the cases here - maybe simplify to:
```suggestion
            if glob == "":
                return pkg_json.root_dir == ""
```

Reply:
Sure!